packages/did/src/DidLinks/AccountLinks.chain.ts

```diff
@@ -59,12 +56,11 @@ export type Address = KiltAddress | SubstrateAddress | EthereumAddress
 /**
  * Type of a linking payload signing function.
  *
- * It takes the HEX-encoded tuple (DidAddress, BlockNumber) and expects the HEX-encoded signature generated by the provided address in return.
+ * It takes the HEX-encoded tuple (DidAddress, BlockNumber) and returns the Uint8Array signature generated by the provided address.
  */
-export type LinkingSignerCallback = (
-  encodedLinkingDetails: HexString,
-  address: SubstrateAddress
-) => Promise<HexString>
+export type LinkingSignCallback = (
+  encodedLinkingDetails: HexString
+) => Promise<Uint8Array>
```

Comment:
It was specified to return a hex string, but that was instantly converted back to bytes, so I changed this. If there’s a reason we need that, I’m ready to undo it.

Reply:
what does sporran return when we sign? is that a reason to make it a string?

Reply:
I think it should be the responsibility of the code talking to Sporran to convert hex to bytes before feeding this into the SDK. Do you agree?